service/src/test/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiControllerAzureDiskTest.java

```diff
@@ -0,0 +1,96 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.UUID;
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+public class ControlledAzureResourceApiControllerAzureDiskTest extends BaseAzureUnitTest {
+  @Autowired MockMvc mockMvc;
+  @Autowired ObjectMapper objectMapper;
+  @Autowired ControlledAzureResourceApiController controller;
+
+  @BeforeEach
+  void setUp() {
+    when(mockSamService()
+            .getUserEmailFromSamAndRethrowOnInterrupt(any(AuthenticatedUserRequest.class)))
+        .thenReturn(DEFAULT_USER_EMAIL);
+  }
+
+  @Test
+  public void createAzureDisk400WithNoParameters() throws Exception {
+    UUID workspaceId = UUID.randomUUID();
+    mockMvc
+        .perform(
+            addAuth(
+                post(String.format(CREATE_CONTROLLED_AZURE_DISK_V2_PATH_FORMAT, workspaceId))
+                    .contentType(MediaType.APPLICATION_JSON_VALUE)
+                    .accept(MediaType.APPLICATION_JSON)
+                    .characterEncoding("UTF-8")
+                    .content("{}"),
+                USER_REQUEST))
+        .andExpect(status().is(HttpStatus.SC_BAD_REQUEST));
+  }
+
+  @Test
+  public void createDisk() throws Exception {
+    UUID workspaceId = UUID.randomUUID();
+    setupMockLandingZoneRegion(Region.US_SOUTH_CENTRAL);
+
+    ApiControlledResourceCommonFields commonFields =
+        ControlledResourceFixtures.makeDefaultControlledResourceFieldsApi();
+
+    ApiAzureDiskCreationParameters params =
+        new ApiAzureDiskCreationParameters().name("disk1").size(100);
+
+    ApiCreateControlledAzureDiskRequestV2Body diskRequestV2Body =
+        new ApiCreateControlledAzureDiskRequestV2Body()
+            .common(commonFields)
+            .azureDisk(params)
+            .jobControl(new ApiJobControl().id(UUID.randomUUID().toString()));
+
+    when(getMockJobApiUtils().retrieveAsyncJobResult(any(), eq(ControlledAzureDiskResource.class)))
+        .thenReturn(
+            new JobApiUtils.AsyncJobResult<ControlledAzureDiskResource>()
+                .jobReport(new ApiJobReport().status(ApiJobReport.StatusEnum.SUCCEEDED)));
+
+    setupMockLandingZoneRegion(Region.GERMANY_CENTRAL);
+
+    mockMvc
+        .perform(
+            addJsonContentType(
+                addAuth(
+                    post(String.format(CREATE_CONTROLLED_AZURE_DISK_V2_PATH_FORMAT, workspaceId))
+                        .content(objectMapper.writeValueAsString(diskRequestV2Body)),
+                    USER_REQUEST)))
+        .andExpect(status().is(HttpStatus.SC_OK))
+        .andExpect(MockMvcResultMatchers.jsonPath("$.jobReport").exists())
+        .andExpect(MockMvcResultMatchers.jsonPath("$.resourceId").exists());
```

Comment:
cc @lucymcnatt , added test coverage for this too so future updates on `CreateControlledAzureResource` won't remove the leo requirement of resourceId

Reply:
nice thanks!